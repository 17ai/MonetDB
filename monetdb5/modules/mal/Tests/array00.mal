#create the dimensions for a few arrays

#create array v(i int dimension[0:1:4])
v:= array.series(0,1,4,1,1);
io.print(v);

#create array m(x int dimension[0:1:3], y int dimension[0:1:3], int default 333);
mx:= array.series(0,1,3,3,1);
my:= array.series(0,1,3,1,3);
mv:= array.filler(mx,333);
io.print(mx,my,mv);

#create array m(x int dimension[0:2:6], y int dimension[0:1:4], int default 333);
mx:= array.series(0,2,6,4,1);
my:= array.series(0,1,4,1,3);
mv:= array.filler(mx,333);
io.print(mx,my,mv);

#create array u(x int dimension[0:*:1]);
#any * implies this one
ux:= array.series(0,0,0,0,0);
io.print(ux);
