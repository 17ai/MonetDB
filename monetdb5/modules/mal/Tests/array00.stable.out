stdout of test 'array00` in directory 'modules/mal` itself:


# 10:51:36 >  
# 10:51:36 >   mserver5  --debug=10 --set gdk_nr_threads=0  --set "gdk_dbfarm=/ufs/mk/sciql//Linux/var/MonetDB" --set mapi_open=true --set mapi_port=38302 --set monet_prompt= --trace --forcemito --set mal_listing=2  --dbname=mTests_modules_mal  array00.mal
# 10:51:36 >  

# MonetDB 5 server v11.4.0
# This is an unreleased version
# Serving database 'mTests_modules_mal', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 7.749 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2011 MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:38302/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
function user.main():void;
#create the dimensions for a few arrays 
#create array v(i int dimension[0:1:4]) 
    v := array.series(0,1,4,1,1);
    io.print(v);
#create array m(x int dimension[0:1:3], y int dimension[0:1:3], int default 333); 
    mx := array.series(0,1,3,3,1);
    my := array.series(0,1,3,1,3);
    mv := array.filler(mx,333);
    io.print(mx,my,mv);
#create array m(x int dimension[0:2:6], y int dimension[0:1:4], int default 333); 
    mx := array.series(0,2,6,4,1);
    my := array.series(0,1,4,1,3);
    mv := array.filler(mx,333);
    io.print(mx,my,mv);
#create array u(x int dimension[0:*:1]); 
#any * implies this one 
    ux := array.series(0,0,0,0,0);
    io.print(ux);
end main;
#-----------------#
# h	t	  # name
# void	int	  # type
#-----------------#
[ 0@0,	  0	  ]
[ 1@0,	  1	  ]
[ 2@0,	  2	  ]
[ 3@0,	  3	  ]
#---------------------------------#
# h	t	t	t	  # name
# void	int	int	int	  # type
#---------------------------------#
[ 0@0,	  0,	  0,	  333	  ]
[ 1@0,	  0,	  1,	  333	  ]
[ 2@0,	  0,	  2,	  333	  ]
[ 3@0,	  1,	  0,	  333	  ]
[ 4@0,	  1,	  1,	  333	  ]
[ 5@0,	  1,	  2,	  333	  ]
[ 6@0,	  2,	  0,	  333	  ]
[ 7@0,	  2,	  1,	  333	  ]
[ 8@0,	  2,	  2,	  333	  ]
#---------------------------------#
# h	t	t	t	  # name
# void	int	int	int	  # type
#---------------------------------#
[ 0@0,	  0,	  0,	  333	  ]
[ 1@0,	  0,	  1,	  333	  ]
[ 2@0,	  0,	  2,	  333	  ]
[ 3@0,	  0,	  3,	  333	  ]
[ 4@0,	  2,	  0,	  333	  ]
[ 5@0,	  2,	  1,	  333	  ]
[ 6@0,	  2,	  2,	  333	  ]
[ 7@0,	  2,	  3,	  333	  ]
[ 8@0,	  4,	  0,	  333	  ]
[ 9@0,	  4,	  1,	  333	  ]
[ 10@0,	  4,	  2,	  333	  ]
[ 11@0,	  4,	  3,	  333	  ]
!MALException:array.series:illegal range



# 10:51:36 >  
# 10:51:36 >  Done.
# 10:51:36 >  

